token/core/zkatdlog/crypto/audit/auditor.go

```diff
@@ -334,6 +334,7 @@ func GetAuditInfoForIssues(issues [][]byte, metadata []driver.IssueMetadata) ([]
 		if len(ia.OutputTokens) != len(md.ReceiversAuditInfos) || len(ia.OutputTokens) != len(md.OutputsMetadata) {
 			return nil, errors.Errorf("number of output does not match number of provided metadata")
 		}
+		outputs[k] = make([]*AuditableToken, len(md.ReceiversAuditInfos))
```

Comment:
if you allocate the array as 
```go
outputs[k] = make([]*AuditableToken, 0, len(md.ReceiversAuditInfos))
```
Then you can keep `append`. The above call creates an array of size len(md.ReceiversAuditInfos) and 0 length.